### PR TITLE
Creating the new routs and entities of sr-preset

### PR DESCRIPTION
**Pull request by AlexanderBelkin**

### CODE REVIEW

File: services/app-server/src/modules/users/users.routes.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const guard = require("express-jwt-permissions")();
 router.post("/authenticate", usersController.authenticate);
 router.post("/register", usersController.register);
 router.use(jwt({ allowed: [] })); // From here all are protected
-router.post("/create", guard.check(["user:write"]), usersController.register);
+router.post("/create", usersController.register);
```
Comment (yonigo) [inline services/app-server/src/modules/users/users.routes.js]: Why did you change this?

This route needs to be secure..

File: services/app-server/src/modules/sr-preset/sr-preset.service.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+require("rootpath")();
+const SrPreset = require("./models/sr-preset.model");
+const ObjectId = require("db-wrapper").mongoose.Types.ObjectId;
+
+async function getAll(_id) {
+  return await SrPreset.find({
+    user: { _id },
```
Comment (yonigo) [inline services/app-server/src/modules/sr-preset/sr-preset.service.js]: We also need to return presets created by any of the ancestors of a user (ancestor list is user.path.split(''/"))

File: services/app-server/src/modules/sr-preset/sr-preset.service.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  }).populate({
+    path: "user",
+    match: { _id: _id },
+    select: "name"
+  });
+}
+
+async function getById(id) {
```
Comment (yonigo) [inline services/app-server/src/modules/sr-preset/sr-preset.service.js]: Same here. only return the preset if its made by the user or an ancestor

File: services/app-server/src/modules/sr-preset/sr-preset.service.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+async function create(srPresetParams, creatingUserId) {
+  srPresetParams.user = creatingUserId;
+  const srPreset = new SrPreset(srPresetParams);
+  await srPreset.save();
+  return Promise.resolve(srPreset);
+}
+
+async function update(id, srPresetParams, userId) {
+  const srPreset = await SrPreset.findById(id);
+  if (!userId) throw "User not found";
```
Comment (yonigo) [inline services/app-server/src/modules/sr-preset/sr-preset.service.js]: Also, if the user is not the creating user, don't allow the update..

Comment (yonigo) [inline services/app-server/src/modules/sr-preset/sr-preset.service.js]: There is no need for the userId check. the jwt route guard will fail before if there is no auth token with a user.

File: services/app-server/src/modules/sr-preset/sr-preset.service.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  Object.assign(srPreset, srPresetParams);
+  await srPreset.save();
+  return Promise.resolve(srPreset);
+}
+
+async function _delete(id, userId) {
+  if (!userId) throw "User not found";
```
Comment (yonigo) [inline services/app-server/src/modules/sr-preset/sr-preset.service.js]: There is no need for the userId check. the jwt route guard will fail before if there is no auth token with a user.

File: services/app-server/src/modules/users/users.routes.js (hunk 1 above)
Comment (AlexanderBelkin) [reply services/app-server/src/modules/users/users.routes.js]: ok I got it
